main.py

```diff
@@ -0,0 +1,515 @@
+                self.rect.left += self.pos + 5
+                if self.rect.left >= screen_width - 48 :                # won't go past edge of screen
+                    self.rect.left = screen_width - 48
+
+            elif self.pos == -1:                                        # if left key is pressed, character moves left
+                self.rect.left += self.pos - 5
+                if self.rect.left <= 0:                                 # won't go past edge of screen
+                    self.rect.left = 0
+
+class Gun(pygame.sprite.Sprite):
+    """ Weapon that shoots arrow to pop bubbles """
+    def __init__(self, x = 0, y = 0):
+        pygame.sprite.Sprite.__init__(self)
+        self.active = False                                             # won't fire unless space key is pressed
+        self.image = pygame.image.load(CURR_DIR + '/images/arrow2.png')
+        self.rect = self.image.get_rect()
+        self.rect.left = x
+        self.rect.top = y
+
+    def update(self):
+        if self.active:
+            if self.rect.top <= 0:                                      # disappears when hits top of screen                      
+                self.active = False
+                self.kill()
+            else:
+                self.rect.top -= 10
+        else:
+            self.kill()
+
+class Ball(pygame.sprite.Sprite):
+    """ Bubbles that character needs to pop """
+    def __init__(self, size, speed, x = 0, y = 0):
+        pygame.sprite.Sprite.__init__(self)
+        self.size = size
+        self.speed = speed
+        self.pic = random.choice(pictures)                              # randomly chooses between given pictures of Olin faces
+        self.image = pygame.image.load(self.pic)
+        self.image = pygame.transform.scale(self.image,(size*10,size*10))
+        self.rect = self.image.get_rect()
+        self.rect.left = x
+        self.rect.top = y
+
+    def update(self):
+        self.rect = self.rect.move(self.speed)
+
+        if self.rect.left < 0 or self.rect.right > screen_width:        # bubbles bounce off of left/right/top/bottom walls
+            self.speed[0] = -self.speed[0]
+        if self.rect.top < 100 or self.rect.bottom > screen_height:
+            self.speed[1] = -self.speed[1]
+
+        self.rect.left = self.walls(self.rect.left, 0 , screen_width)
+        self.rect.right = self.walls(self.rect.right, 0 , screen_width)
+        self.rect.top = self.walls(self.rect.top, 100 , screen_height)
+        self.rect.bottom = self.walls(self.rect.bottom, 0 , screen_height)
+
+        self.speed[1] += gravity                                        # simulates effect of gravity
+
+    def walls(self,number, min_number, max_number):
+        """ helper function for ball to bounce off of walls """
+        self.number = number
+        self.min_number = min_number
+        self.max_number = max_number
+        return min(max(self.number,self.min_number),self.max_number)
```

Comment:
An in-line comment here about what the `min` and `max` are for would be helpful.